flutter_local_notifications_platform_interface/pubspec.yaml

```diff
@@ -1,15 +1,13 @@
 name: flutter_local_notifications_platform_interface
 description: A common platform interface for the flutter_local_notifications plugin.
-version: 7.2.0
+version: 7.3.0
 homepage: https://github.com/MaikuB/flutter_local_notifications/tree/master/flutter_local_notifications_platform_interface
 
 environment:
   sdk: ^3.1.0
   flutter: ">=3.13.0"
 
 dependencies:
-  flutter:
```

Comment:
Just noticed this but is there a reason this got removed? Could it be restored? Whilst the `plugin_platform_interface` itself doesn't depend on Flutter, I can see the platform interfaces for other plugins have this included. The [PR](https://github.com/MaikuB/flutter_local_notifications/pull/2393/files) I have includes a fix to have API docs reference `PlatformException` and this required importing `'package:flutter/services.dart'`. Could be it still functions without it but IMO looks off to not include given the import and how this is for Flutter

Reply:
Turns out, because I used FFI and not Flutter's Method Channels, the Windows version of this plugin is fully available for non-Flutter projects, and can be used in CLI apps (eg, start processing a file, send a notification when done). I think this is a valuable use-case and there are no plugins available on Pub to serve this need (and if there were one, it would ideally use FFI in the same way I did here). 

To allow this, I had to remove `flutter` as a dependency on the `_windows` plugin, and that means transitively removing it here as well. It doesn't impact functionality at all, but would instead allow even more developers to use this package. And the same could be done for Linux, since it is also FFI-based. If the MacOS version is re-implemented in FFI in the future it would benefit from this as well.

https://github.com/MaikuB/flutter_local_notifications/blob/d565daab613052fecaf723df865101b46ee5a26a/flutter_local_notifications_platform_interface/lib/flutter_local_notifications_platform_interface.dart#L87-L93

Reading these comments, it seems that it's basically the same as an `UnsupportedError`, which is unfortunately an `Error`, and not an `Exception`. `PlatformException` does seem like the right tool for the job, but I really don't think including the DartDoc link is worth totally breaking support for CLI apps on Windows and Linux. Maybe just link it the regular way?
```dart
/// [`PlatformException`](https://api.flutter.dev/flutter/services/PlatformException-class.html)
```

Reply:
Gotcha and makes. I forgot the markdown approach for links could be used for links as well. Thanks